### PR TITLE
[GithubAction] Push Operator Node Images

### PR DESCRIPTION
**Pull request by siddimore**
## Why are these changes needed?
Needed to push operator node images to public ghcr. 
Invoking Role: Oncall
Steps: 
1. Verify Test Results for PreProd run and obtain commit-sha for that (TBD: one place look up for commit-sha to TestResult)
2. Provide Commit Sha and Release Tag as input parameters to this action
3. Run workflow

## Alternative Considered:
1. Writing this github action in eigenda-devops 
2. PreProd environment rel has a github action(cronjob) which will publish eigenda components update and run some test
3. If this PreProd environment rel succeeds then publish Node specific images from ECR to GHCR 

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/workflows/docker-publish-opr-node-images.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+      if: ${{ success() }}
+
+    - name: Set Tag Name
+      id: set_tag
+      run: echo "::set-output name=tag::${{ github.event.inputs.release_tag || github.event.inputs.commit_sha }}"
```
Comment (dmanc) [inline .github/workflows/docker-publish-opr-node-images.yaml:54]: How does this work? I see below that the tag is ${{ github.event.inputs.commit_sha }} but don't we also want the release tag?

Comment (dmanc) [reply .github/workflows/docker-publish-opr-node-images.yaml:54]: Should it be ${{ steps.set_tag.outputs.tag }}?

Comment (siddimore) [reply .github/workflows/docker-publish-opr-node-images.yaml:54]: good catch....meant to use that

Comment (mpjunior92) [inline .github/workflows/docker-publish-opr-node-images.yaml:54]: `::set-output` is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples)

File: .github/workflows/docker-publish-opr-node-images.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      if: ${{ success() }}
+
+    - name: Build and Push Opr Node Image
+      uses: docker/build-push-action@v2
+      with:
+        file: ./node/cmd/Dockerfile
+        push: true
+        tags: ${{ env.REGISTRY }}/layr-labs/eigenda/opr-node:${{ steps.set_tag.outputs.tag }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache-new
```
Comment (mpjunior92) [inline .github/workflows/docker-publish-opr-node-images.yaml:64]: `/tmp/.buildx-cache-new` is a good candidate to be a env var

**Review by mpjunior92 — APPROVED** [review]

**Review by dmanc — APPROVED** [review]
Let's do a test run before merging

**Pull request merged**